### PR TITLE
Integration tests - the battle continues

### PR DESCRIPTION
**Pull request by corylanou**
Mostly aggregation testing this time.  Have several `FIXME` that have to be addressed.

- Fixed error message regressions
- Moved helper/convenience code to it's own file
- Aggregation Tests
- User Tests

### CODE REVIEW

File: tsdb/tx.go [hunk 1]
```diff
@@ -316,7 +316,14 @@ func (l *LocalMapper) Begin(c *influxql.Call, startingTime int64, chunkSize int)
 	if fieldName != "" {
 		fid, err := l.decoder.FieldIDByName(fieldName)
 		if err != nil {
-			return fmt.Errorf("%s isn't a field on measurement %s", fieldName, l.job.MeasurementName)
+			switch {
+			case c.Name == "distinct":
```
Comment (otoolep) [inline tsdb/tx.go]: Is this really necessary? Why richer messages fro `distinct` and `count distinct`?

I mean, I like helpful messages for the user, I'm just not sure why these are special-cased.

Comment (corylanou) [inline tsdb/tx.go]: None of the rich messages are necessary, but they are nice, and already added.  Are you suggesting we take them out?

Comment (otoolep) [inline tsdb/tx.go]: No saying we take them out, but when I see stuff special-cased, I just wondered why certain aggregates are handled nicely, and others aren't.

It's definitely no harm, and obviously adds to it. Not a blocker.

**Comment by otoolep** [comment]
+1

**Pull request merged**